Author: Alessio Treglia <quadrispro@ubuntu.com>
Description: Add device entry for
 Creative ZEN MicroPhoto 041E:4133
 Apple iPhone 05ac:0x1290
 Apple iPod Touch 1st Gen 05ac:1291
 Apple iPhone 3G 05ac:0x1292
 Apple iPod Touch 2nd Gen 05ac:0x1293
 Apple iPhone 3GS 05ac:0x1294
 Apple 05ac:1296
 Apple 05ac:1297
 Apple 05ac:1298
 Apple iPod Touch 3rd Gen 05ac:0x1299
 Apple iPad 05ac:129a
 SonyEricsson U5 0fce:0133
 Samsung Galaxy S GT-I9000 04e8:6877
 Samsung Vibrant SGH-T959 04e8:68a9
 COBY MP705 1e74:6512
 Cowon iAudio J3 0e21:0921
Origin: upstream, commit:1136,1148,1149,1150,1151
Forwarded: http://sourceforge.net/support/tracker.php?aid=3053793
Bug-Ubuntu: https://launchpad.net/bugs/537386
Bug-Debian: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=594560
---
 src/music-players.h |   35 +++++++++++++++++++++++++++++++++++
 1 file changed, 35 insertions(+)

--- libmtp.orig/src/music-players.h
+++ libmtp/src/music-players.h
@@ -52,6 +52,7 @@
   { "Creative", 0x041e, "ZEN Micro (MTP mode)", 0x4130, DEVICE_FLAG_BROKEN_MTPGETOBJPROPLIST_ALL },
   { "Creative", 0x041e, "ZEN Touch (MTP mode)", 0x4131, DEVICE_FLAG_BROKEN_MTPGETOBJPROPLIST_ALL },
   { "Dell", 0x041e, "Dell Pocket DJ (MTP mode)", 0x4132, DEVICE_FLAG_BROKEN_MTPGETOBJPROPLIST_ALL },
+  { "Creative", 0x041e, "ZEN MicroPhoto", 0x4133, DEVICE_FLAG_BROKEN_MTPGETOBJPROPLIST_ALL },
   { "Creative", 0x041e, "ZEN Sleek (MTP mode)", 0x4137, DEVICE_FLAG_BROKEN_MTPGETOBJPROPLIST_ALL },
   { "Creative", 0x041e, "ZEN MicroPhoto", 0x413c, DEVICE_FLAG_BROKEN_MTPGETOBJPROPLIST_ALL },
   { "Creative", 0x041e, "ZEN Sleek Photo", 0x413d, DEVICE_FLAG_BROKEN_MTPGETOBJPROPLIST_ALL },
@@ -234,6 +235,13 @@
   // Added by Greg Fitzgerald <netzdamon@gmail.com>
   { "Samsung", 0x04e8, "SAMSUNG Trance", 0x6763, DEVICE_FLAG_UNLOAD_DRIVER |
       DEVICE_FLAG_NO_ZERO_READS | DEVICE_FLAG_PLAYLIST_SPL_V1 },
+  // From: Erik Berglund <erikjber@users.sourceforge.net>
+  // Logs indicate this needs DEVICE_FLAG_NO_ZERO_READS
+  { "Samsung", 0x04e8, "Galaxy S GT-I9000", 0x6877, DEVICE_FLAG_UNLOAD_DRIVER |
+      DEVICE_FLAG_NO_ZERO_READS | DEVICE_FLAG_PLAYLIST_SPL_V1 },
+  // From: John Gorkos <ab0oo@users.sourceforge.net> and Akos Maroy <darkeye@users.sourceforge.net>
+  { "Samsung", 0x04e8, "Vibrant SGH-T959", 0x68a9, DEVICE_FLAG_UNLOAD_DRIVER |
+      DEVICE_FLAG_NO_ZERO_READS | DEVICE_FLAG_PLAYLIST_SPL_V1 },
   // From anonymous Sourceforge user
   { "Samsung", 0x04e8, "GT-S5230", 0xe20c, DEVICE_FLAG_NONE },
 
@@ -873,6 +881,10 @@
   { "Cowon", 0x0e21, "iAudio 9 (MTP mode)", 0x0911,
    DEVICE_FLAG_UNLOAD_DRIVER | DEVICE_FLAG_BROKEN_MTPGETOBJPROPLIST |
    DEVICE_FLAG_OGG_IS_UNKNOWN | DEVICE_FLAG_FLAC_IS_UNKNOWN },
+  // Reported by Franck VDL <franckv@users.sourceforge.net>
+  { "Cowon", 0x0e21, "iAudio J3 (MTP mode)", 0x0921,
+   DEVICE_FLAG_UNLOAD_DRIVER | DEVICE_FLAG_BROKEN_MTPGETOBJPROPLIST |
+   DEVICE_FLAG_OGG_IS_UNKNOWN | DEVICE_FLAG_FLAC_IS_UNKNOWN },
 
   /*
    * Insignia, dual-mode.
@@ -1039,6 +1051,9 @@
   { "SonyEricsson", 0x0fce, "W705/W715", 0x0105, DEVICE_FLAG_BROKEN_MTPGETOBJPROPLIST },
   // Reported by Håkan Kvist
   { "SonyEricsson", 0x0fce, "W995", 0x0112, DEVICE_FLAG_BROKEN_MTPGETOBJPROPLIST },
+  // Reported by anonymous SourceForge user
+  { "SonyEricsson", 0x0fce, "U5", 0x0133, DEVICE_FLAG_BROKEN_MTPGETOBJPROPLIST },
+
   // Reported by Jonas Nyrén <spectralmks@users.sourceforge.net>
   { "SonyEricsson", 0x0fce, "W302", 0x10c8, DEVICE_FLAG_BROKEN_MTPGETOBJPROPLIST },
 
@@ -1143,6 +1158,26 @@
   { "Nextar", 0x0402, "MA715A-8R", 0x5668 },
 
   /*
+   * Coby
+   */
+  { "Coby", 0x1e74, "COBY MP705", 0x6512 },
+
+  /*
+   * Apple devices, which are not MTP natively but can be made to speak MTP
+   * using PwnTunes (http://www.pwntunes.net/)
+   */
+  { "Apple", 0x05ac, "iPhone", 0x1290 },
+  { "Apple", 0x05ac, "iPod Touch 1st Gen", 0x1291 },
+  { "Apple", 0x05ac, "iPhone 3G", 0x1292 },
+  { "Apple", 0x05ac, "iPod Touch 2nd Gen", 0x1293 },
+  { "Apple", 0x05ac, "iPhone 3GS", 0x1294 },
+  { "Apple", 0x05ac, "0x1296", 0x1296 },
+  { "Apple", 0x05ac, "0x1297", 0x1297 },
+  { "Apple", 0x05ac, "0x1298", 0x1298 },
+  { "Apple", 0x05ac, "iPod Touch 3rd Gen", 0x1299 },
+  { "Apple", 0x05ac, "iPad", 0x129a },
+
+  /*
    * Other strange stuff.
    */
   { "Isabella", 0x0b20, "Her Prototype", 0xddee, DEVICE_FLAG_NONE }
